File: LeetCode/55. 跳跃游戏.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        int mostRight = 0;
        for (int i = 0; i < n;++i) {
            if (i <= mostRight) {
                mostRight = Math.max(mostRight, i + nums[i]);
                if (mostRight >= n -1) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: LeetCode/1-100/55. 跳跃游戏.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function (nums) {
  const n = nums.length
  let mostRight = 0
  for (let i = 0; i < n; ++i) {
    if (mostRight >= i) {
      mostRight = Math.max(mostRight, i + nums[i])
      if (mostRight >= n - 1) return true
    }
  }
  return false
}
